package.json

```diff
@@ -8,32 +8,32 @@
   "author": "Shopify <app-extensions@shopify.com>",
   "license": "MIT",
   "devDependencies": {
-    "@shopify/argo-run": "^0.2.3",
+    "@shopify/argo-run": "^0.3.0-0",
     "@types/prettier": "^2.1.5",
     "@types/yargs": "^15.0.9",
-    "@typescript-eslint/eslint-plugin": "^4.6.0",
-    "@typescript-eslint/parser": "^4.6.0",
-    "eslint": "^7.12.1",
-    "eslint-config-prettier": "^6.15.0",
-    "eslint-plugin-prettier": "^3.1.4",
-    "eslint-plugin-react": "^7.21.5",
+    "@typescript-eslint/eslint-plugin": "^4.15.0",
+    "@typescript-eslint/parser": "^4.15.0",
+    "eslint": "^7.19.0",
+    "eslint-config-prettier": "^7.2.0",
+    "eslint-plugin-prettier": "^3.3.0",
+    "eslint-plugin-react": "^7.22.0",
     "fs-extra": "^9.0.1",
     "inquirer": "^7.3.3",
     "prettier": "^2.1.2",
     "ts-node": "^9.0.0",
-    "typescript": "^3.9.7",
+    "typescript": "^4.1.0",
     "yargs": "^16.1.0"
   },
   "scripts": {
     "generate": "ts-node ./scripts/generate",
     "build": "argo-run build",
     "server": "yarn start",
     "start": "argo-run serve --port 39351",
-    "lint": "eslint 'src/**'"
+    "lint": "eslint 'src/**'",
+    "type-check": "tsc --pretty"
   },
   "dependencies": {
-    "@shopify/argo-checkout": "^0.8.0",
-    "@shopify/argo-checkout-react": "^0.8.0",
-    "react": "^16.13.0"
```

Comment:
If an extension author wants to use e.g. `React.createContext`, would they install `@remote-ui/mini-react` and import that from there? 

If that's the case, wouldn't it be better if `argo-checkout` declared this as `peerDependency` and consumers install it as dependency? Otherwise they'd end up installing `@remote-ui/mini-react` in a newer version than currently specified in `argo-checkout` which could lead to issues(?)

Reply:
If a project wants React, it still imports it as it does now (although the default import is discouraged because it hurts tree shaking:

```ts
import {createContext} from 'react';
```

At build/ dev time, we swap that out with an import for `@remote-ui/mini-react/compat`. This is basically the same strategy as Preact's compatibility layer, and it means that even though the app code (and our own argo-checkout-react source code) still write against the React package/ types, that exact package can be switched out transparently and does not need to be installed as a dependency. `remote-ui/mini-react` is a dependency of `argo-checkout-react`, so we don't have to worry about it not being installed.

Reply:
Yeah that makes sense. Should we keep react as `devDependency` then so that when they require it, the editor doesn't complain?

Reply:
Did you get TypeScript errors when importing React? The build errors from a missing react should be resolved with the latest branch, now that the real argo-checkout packages are published and used here, and TypeScript seems fine with a react import on my branch.

Reply:
Seems to work fine. I just wonder if that'd be the case once people use https://github.com/benmosher/eslint-plugin-import as `react` isn't a top level dependency.

But I guess, they'd install it themselves in that case.

Reply:
We install that by default and it appears to work OK, but definitely on the lookout for incompatibilities